tests/integration/pilot/locality/failover/failover_test.go

```diff
@@ -0,0 +1,235 @@
+//
+// The created Service Entry for fake-(cds|eds)-external-service-12345.com points the domain at services
+// that exist internally in the mesh. In the Service Entry we set a non-existent service or IP to be in the same
+// locality as the caller (A). Service B to be in the primary failover locality and service C to be in the secondary failover locality.
+// For CDS, the endpoint pool for the service in the local locality is empty so it fails over to service B.
+// For EDS, when the request to the local locality fails outlier detection ejects the fake IP and fails over to service B.
+// We then verify that all request go to service B. For details check the Service Entry configuration at the bottom of the page.
+//
+//  CDS Test
+//
+//                                                            100% +-> b (closeregion.zone.subzone)
+//                                                                 |
+//                                                                 |
+//                                                                 |
+// A (region.zone.subzone) -> fake-cds-external-service-12345.com -|-> c (notcloseregion.zone.subzone)
+//                                                                 |
+//                                                                 |
+//                                                                 |
+//                                                                 +-> NonExistentService (region.zone.subzone)
+//
+//
+//  EDS Test
+//
+//                                                            100% +-> 10.28.1.138 (b -> closeregion.zone.subzone)
+//                                                                 |
+//                                                                 |
+//                                                                 |
+// A (region.zone.subzone) -> fake-eds-external-service-12345.com -|-> 10.28.1.139 (c -> notcloseregion.zone.subzone)
+//                                                                 |
+//                                                                 |
+//                                                                 |
+//                                                                 +-> 10.10.10.10 (NonExistentService -> region.zone.subzone)
+//
+
+const (
+	testDuration = 10 * time.Second
+)
+
+var serviceBHostname = regexp.MustCompile("^b-.*$")
+
+type ServiceConfig struct {
+	Name               string
+	Host               string
+	Namespace          string
+	Resolution         string
+	ServiceBAddress    string
+	ServiceCAddress    string
+	NonExistantService string
+}
+
+func TestLocalityFailover(t *testing.T) {
+	ctx := framework.NewContext(t)
+	defer ctx.Done(t)
+
+	ctx.RequireOrSkip(t, environment.Kube)
+	ctx.Environment().(*kube.Environment).Settings()
+
+	// Share Istio Control Plane
+	g := galley.NewOrFail(t, ctx, galley.Config{})
+	p := pilot.NewOrFail(t, ctx, pilot.Config{Galley: g})
+
+	rand.Seed(time.Now().UnixNano())
```

Comment:
nit could just use CDs/eds suffix instead of random number if you wanted? doesn't matter at all though

Reply:
This is to future proof it from potential hostname conflicts in the same cluster. I don't know how likely we are to hit it but this protects us from that.